### PR TITLE
Tiles download time greatly improved

### PR DESCRIPTION
**Pull request by helgasoft**
changed some functions to _async_ and removed recursion from __loadTile_
Now tiles load asynchronously 'in parallel' just in a few seconds.  Here is a [live demo](https://gistpreview.github.io/?5e3c8e85339c188d3e300b3d08ce3e85).

PPS:  I love all the improvements in v.2  (iDB, promises, api) - way to go!

### CODE REVIEW

**Comment by helgasoft** [comment]
SaveTilesTest.js (for Travis) has a line

> assert(stub.**calledThrice**, '_loadTile has not been called');

Apparently the assumption is that _loadTile should be called 3 times - for subdomains a,b,c. 
I rewrote _loadTile as non-recurrent and it is called many more times. This is the reason this test is now failing.

**Comment by allartk** [comment]
Looks good to me! 

Does 
assert.isAtLeast(stub.callCount,3);

fix the test?

**Comment by helgasoft** [comment]
Change to _assert.isAtLeast_ does not fix the test. I suspect it's related to async execution. Will investigate further...

File: README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # leaflet.offline version 2.x
 
 [![npm version](https://badge.fury.io/js/leaflet.offline.svg)](https://badge.fury.io/js/leaflet.offline)
-[![Build Status](https://travis-ci.org/allartk/leaflet.offline.svg?branch=travis)](https://travis-ci.org/allartk/leaflet.offline)
+[![Build Status](https://travis-ci.org/helgasoft/leaflet.offline.png?branch=master)](https://travis-ci.org/helgasoft/leaflet.offline)
```
Comment (allartk) [inline README.md]: could you undo this change before merge? :)

Comment (helgasoft) [reply README.md]: just did.  If you need further changes, please feel free to do them -  _Allow edits from maintainers_ is enabled.
